src/modules/navigator/mission.cpp

```diff
@@ -1512,3 +1569,44 @@ Mission::generate_waypoint_from_heading(struct position_setpoint_s *setpoint, fl
 	setpoint->type = position_setpoint_s::SETPOINT_TYPE_POSITION;
 	setpoint->yaw = yaw;
 }
+
+uint16_t
+Mission::index_closest_mission_item() const
+{
+	uint16_t min_dist_index(0);
+	float min_dist(FLT_MAX), dist_xy(FLT_MAX), dist_z(FLT_MAX);
+
+	dm_item_t dm_current = (dm_item_t)(_offboard_mission.dataman_id);
+
+	for (size_t i = 0; i < _offboard_mission.count; i++) {
+		struct mission_item_s missionitem = {};
+		const ssize_t len = sizeof(missionitem);
+
+		if (dm_read(dm_current, i, &missionitem, len) != len) {
+			/* not supposed to happen unless the datamanager can't access the SD card, etc. */
+			PX4_ERR("dataman read failure");
+			break;
+		}
+
+		if (item_contains_position(missionitem)) {
+			// do not consider land waypoints for a fw
```

Comment:
> do not consider land waypoints for a fw

Why not?

Reply:
The reason is that for a fw we must consider the landing approach. If we would consider the landing waypoint as a closest waypoint we could approach the landing strip with a completely different approach than planned which could result in a crash.

Reply:
yep.